ingest/ingest-datasets.py

```diff
@@ -0,0 +1,130 @@
+    # graph = describe_object_function(endpoint=endpoint, dataset=dataset)
+
+    ds = graph.resource(dataset)
+
+    try:
+        title = ds.label().toPython()
+    except AttributeError:
+        print("missing title:", dataset)
+        return {}
+
+    dco_id = list(ds.objects(DCO.hasDcoId))
+    dco_id = str(dco_id[0].identifier) if dco_id else None
+
+    doc = {"uri": dataset, "title": title, "dcoId": dco_id}
+
+    doi = list(ds.objects(BIBO.doi))
+    doi = doi[0].toPython() if doi else None
+    if doi:
+        doc.update({"doi": doi})
+
+    abstract = list(ds.objects(BIBO.abstract))
+    abstract = abstract[0].toPython() if abstract else None
+    if abstract:
+        doc.update({"abstract": abstract})
+
+    most_specific_type = list(ds.objects(VITRO.mostSpecificType))
+    most_specific_type = most_specific_type[0].label().toPython() \
+        if most_specific_type and most_specific_type[0].label() \
+        else None
+    if most_specific_type:
+        doc.update({"mostSpecificType": most_specific_type})
+
+    publication_year = list(ds.objects(DCO.yearOfPublication))
+    publication_year = publication_year[0] if publication_year else None
+    if publication_year:
+        doc.update({"publicationYear": str(publication_year)})
+
+    dco_communities = get_dco_communities(ds)
+    if dco_communities:
+        doc.update({"dcoCommunities": dco_communities})
+
+    # portal_groups
+    portal_groups = get_portal_groups(ds)
+    if portal_groups:
+        doc.update({"portalGroups": portal_groups})
+
+    # projects
+    projects = get_projects_of_dataset(ds)
+    if projects:
+        doc.update({"projects": projects})
+
+    # dataType
+    data_types = get_data_types(ds)
+    if data_types:
+        doc.update({"dataTypes": data_types})
+
+    # cites
+    cites = get_cites(ds)
+    if cites:
+        doc.update({"citations": cites})
+
+    # authors: if none, will return an empty list []
+    authors = get_authors(ds)
+    doc.update({"authors": authors})
+
+    # distributions: if none, will return an empty list []
+    distributions = get_distributions(ds)
+    doc.update({"distributions": distributions})
+
+    return doc
+
+
+
+if __name__ == "__main__":
+    # Usage:
+    #   Arguments:
+    #   --threads: number of threads to use (default = 8)
+    #   --es', elasticsearch service URL (default="http://localhost:9200")
+    #   --publish', publish to elasticsearch? (default=False)
+    #   --rebuild', rebuild elasticsearch index? (default=False)
+    #   --mapping', dataset elasticsearch mapping document (default="mappings/dataset.json")
+    #   --sparql', sparql endpoint (default='http://deepcarbon.tw.rpi.edu:3030/VIVO/query')
+    #   [out]: file name of the elasticsearch bulk ingest file
+    # Example:
+    #   python3 ingest-datasets.py output4 --threads 2 --mapping mappings/XXXXX.json ...
+
+    Main(get_objects_query=get_datasets_query, create_object_doc_function=create_dataset_doc,
```

Comment:
so rename class Main to DatasetIngest (or similar)?

My preference would be to refactor the argument parsing out of the DatasetIngest class.